lib/oauth2/client.ex

```diff
@@ -202,6 +207,43 @@ defmodule OAuth2.Client do
     url
   end
 
+  @doc """
+  Register a serialization module for a given mime type.
+
+  ## Example
+
+      iex> client = OAuth2.Client.put_serializer(%OAuth2.Client{}, "application/json", Jason)
+      %OAuth2.Client{serializers: %{"application/json" => Jason}}
+      iex> OAuth2.Client.get_serializer(client, "application/json")
+      Jason
+  """
+  @spec put_serializer(t, binary, atom) :: t
+  def put_serializer(%Client{serializers: serializers} = client, mime, module)
+    when is_binary(mime) and is_atom(module) do
+    %Client{client | serializers: Map.put(serializers, mime, module)}
+  end
+
+  @doc """
+  Un-register a serialization module for a given mime type.
+
+  ## Example
+
+      iex> client = OAuth2.Client.delete_serializer(%OAuth2.Client{}, "application/json")
+      %OAuth2.Client{}
+      iex> OAuth2.Client.get_serializer(client, "application/json")
+      OAuth2.Serializer.Null
+  """
+  @spec delete_serializer(t, binary) :: t
+  def delete_serializer(%Client{serializers: serializers} = client, mime) do
+    %Client{client | serializers: Map.delete(serializers, mime)}
+  end
+
+  @doc false
+  @spec get_serializer(t, binary) :: atom
+  def get_serializer(%Client{serializers: serializers}, mime) do
+    Map.get(serializers, mime) || OAuth2.Serializer.Null
```

Comment:
See above comment RE: `OAuth2.Serializer.Null` vs `nil`.

Reply:
Could you use `Map.get/3` here instead?

Reply:
@yordis yep, very true. I'm probably going to end up removing the `Null` serializer altogether.

Reply:
Thanks for the update @scrogson 😁